### PR TITLE
[FeatureHighlight] Add missing test for Dynamic Type and mark mdc_legacyFontScaling deprecated.

### PR DESCRIPTION
**Pull request by fluxminds**
Part of https://github.com/material-components/material-components-ios/issues/8304.

### CODE REVIEW

**Comment by jverkoey** [comment]
The pr looks like it’s adding a new api; please update the PR description accordingly.

File: components/FeatureHighlight/src/MDCFeatureHighlightViewController.h [hunk 1]
```diff
@@ -150,7 +150,19 @@ typedef void (^MDCFeatureHighlightCompletion)(BOOL accepted);
  Enable legacy font scaling curves for Dynamic Type.
  Default value is NO.
  */
-@property(nonatomic, readwrite, setter=mdc_setLegacyFontScaling:) BOOL mdc_legacyFontScaling;
+@property(nonatomic, readwrite, setter=mdc_setLegacyFontScaling:)
+    BOOL mdc_legacyFontScaling __deprecated;
```
Comment (romoore) [inline components/FeatureHighlight/src/MDCFeatureHighlightViewController.h:154]: Please use `__deprecated_msg("Use <new API> instead")` to mark deprecated APIs that have replacements.  In general, the project's convention is to provide a helpful message to clients using a deprecated API.

Comment (romoore) [inline components/FeatureHighlight/src/MDCFeatureHighlightViewController.h:154]: Should this pass-through to the new API? That way they both have the same values? If not, it seems like the interaction between the two is non-trivial.

If `mdc_legacyFontScaling` is `YES` but `adjustsFontForContentSizeCategoryWhenScaledFontIsUnavailable` is `NO`, should you use the fall-back scaling or not?

File: components/FeatureHighlight/tests/unit/FeatureHighlightViewControllerTests.m [hunk 2]
```diff
@@ -69,4 +87,40 @@ - (void)testTraitCollectionDidChangeBlockCalledWithExpectedParameters {
   XCTAssertEqual(passedTraitCollection, fakeTraitCollection);
 }
 
+- (void)testFeatureHighlightViewControllerDynamicTypeBehavior {
```
Comment (romoore) [inline components/FeatureHighlight/tests/unit/FeatureHighlightViewControllerTests.m:90]: Please add tests that assert the behavior of the intersection of the two APIs.  It's not clear to me how they would work together given their independent storage in the current version of this PR.

| adjustsFontForContentSizeCategoryWhenScaledFontIsUnavailable | mdc_legacyFontScaling | Font Scaling Type|
|---|---|---|
|N | N | N |
|N | Y | ?? |
|Y | N | ?? |
|Y | Y | Y |

Comment (romoore) [inline components/FeatureHighlight/tests/unit/FeatureHighlightViewControllerTests.m:90]: Please add a test for the new API's default value.

File: components/FeatureHighlight/src/private/MDCFeatureHighlightView+Private.m [hunk 3]
```diff
@@ -215,19 +215,19 @@ - (void)updateBodyFont {
...
     _bodyFont = [MDCFeatureHighlightView defaultBodyFont];
   }
+  UIFont *bodyFont = _bodyFont;
   if (_mdc_adjustsFontForContentSizeCategory) {
-    if (_bodyFont.mdc_scalingCurve && !_mdc_legacyFontScaling) {
+    if (bodyFont.mdc_scalingCurve && !_mdc_legacyFontScaling) {
       // The font has an associated curve (M2+)
-      _bodyLabel.font = [_bodyFont mdc_scaledFontForCurrentSizeCategory];
-    } else {
+      bodyFont = [bodyFont mdc_scaledFontForTraitEnvironment:self];
```
Comment (romoore) [inline components/FeatureHighlight/src/private/MDCFeatureHighlightView+Private.m:222]: Should the font be set twice here [due to the way `UILabel` attempts to optimize `UIFont` setting](https://github.com/material-components/material-components-ios/issues/7424)?

**Review by romoore — CHANGES_REQUESTED** [review]
This PR looks like two semi-related changes based on how I'm reading it.  I believe it should be split into two PRs.

1.  Introduce the new `adjustsFontForContentSizeCategoryWhenScaledFontIsUnavailable` API and deprecate `mdc_legacyFontScaling`. I'm concerned that they are currently not backed by the same instance variable and the interaction of the two APIs is confusing based on how I read the public header documentation. This may just be a matter of documentation, but it will help me to see the unit tests to better understand the behavior.
2. It changes the M2 scaling curves to use `mdc_scaledFontForTraitEnvironment:self` rather than `mdc_scaledFontForCurrentSizeCategory`. While this is a rather minor change, it is not necessary for the new API and if the other change needs to be rolled-back, this refactoring would be lost.

**Comment by jverkoey** [comment]
Closing this due to inactivity, but please reopen if we intend to pursue it again.

**Pull request closed without merging**